### ISSUE TITLE
Make "No changes. Your infrastructure matches the configuration." clearer

### ISSUE THREAD

**Issue by appilon** [hashicorp/terraform#30502]
### Current Terraform Version

```
1.0.3
```
I have personally experienced, and noticed in a provider issue that when a plan detects upstream drift and presents it to the user, there is some confusion in understanding that in fact, everything is fine, and that terraform is suggesting a `terraform apply -refresh-only` to your benefit, but there is no change detected in the user's configuration (this comes up with drift of computed attributes).
### Proposal

Perhaps pull the messaging that no changes have been detected higher in the output so it's the first thing the user sees.
```
No changes. Your infrastructure matches the configuration.

Your configuration already matches the changes detected above. If you'd like to update the Terraform state to match, create and apply a refresh-only plan:  terraform apply -refresh-only
────────────────────────────────────────────────────────────
```

Instead of leading off with a diff, which I think instinctively makes the user think changes/potential destruction will take place
```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # vsphere_virtual_machine.vm has changed
  ~ resource "vsphere_virtual_machine" "vm" {
      ~ default_ip_address                      = "172.16.11.202" -> "172.16.11.192"
      ~ guest_ip_addresses                      = [
          - "172.16.11.202",
          + "172.16.11.192",
            "fe80::250:56ff:fe82:5a9a",
        ]
        id                                      = "4202d9f3-d384-35cf-9c49-a2934e6f3b4f"
        name                                    = "ubuntu-cloud-init"
        tags                                    = []
        # (64 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }


Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or respond to these changes.
```

### References

https://github.com/hashicorp/terraform-provider-vsphere/issues/1463

**Comment by appilon** [comment]
I agree that ultimately this should be a data driven decision and had not considered that "above the hold" situation.

**Comment by rlabrecquefsg** [comment]
I've run into this quite a bit myself, and I think it's specifically this part of this line in particular that bothers me.

> Unless you have made equivalent changes to your configuration, **or ignored the relevant attributes using ignore_changes**, the following plan may include actions to undo or respond to these changes

I feel like Terraform should already know that that the attributes it's actively showing me are in ignore_changes (or just are "computed values" in which case I can't even make the equivalent changes to my configuration???). In this specific case I'd love for it not to lump that in with the exact same output where real changes may have happened. I've already said that explicitly stated via ignore_changes that don't care about it, update the terraform state, sure, but I hardly even really want to know it happened... Hide the current view behind a config flag.

I can not expect all of the developers using the Terraform configuration files I stood up to know whether or not the changes are ignored or not. Hell I can't even remember myself anymore, so what's actionable with those attributes getting displayed as is? Ideally we'd go "Oh wow, that's unexpected that the database scaled way down outside of Terraform!", but instead that gets lost amongst dozens of other computed values that we explicitly never care about, and we basically just ignore the all of the output from that section...

(Note that this is more general than just when you've made no local changes.)
